src/App.tsx

```diff
@@ -2,24 +1,62 @@
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [total, setTotal] = useState(items.length);
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const handlePageChange = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+    const target = e.target as HTMLAnchorElement;
+    const value = target.getAttribute('href')?.substring(1);
+
+    setCurrentPage(prevPage => {
+      switch (value) {
+        case 'prev': {
+          return prevPage - 1;
+        }
+
+        case 'next': {
+          return prevPage + 1;
+        }
+
+        default: {
+          return Number(value);
+        }
+      }
+    });
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${(currentPage - 1) * perPage + 1} - ${currentPage * perPage > total ? total : currentPage * perPage} of ${total})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            value={perPage}
+            onChange={e => {
+              setPerPage(() => {
+                setCurrentPage(1);
+
+                return Number(e.target.value);
```

Comment:
The `setPerPage` function should directly update the `perPage` state. Consider simplifying this by directly setting the new value without using a function inside `setPerPage`.